hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -408,10 +411,10 @@ public void run() throws IOException {
    * An RpcCallBack that creates a list of scanners that needs to perform callBack operation on
    * completion of multiGets.
    */
-   static class RegionScannersCloseCallBack implements RpcCallback {
+  public static class RegionScannersCloseCallBack implements RpcCallback {
     private final List<RegionScanner> scanners = new ArrayList<>();
 
-    public void addScanner(RegionScanner scanner) {
+    public synchronized void addScanner(RegionScanner scanner) {
```

Comment:
Why we need this now?

Reply:
Because it may be called concurrently in different GetActionHandler.